network/dag/pal.go

```diff
@@ -0,0 +1,88 @@
+package dag
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/network/log"
+	"github.com/nuts-foundation/nuts-node/vdr/types"
+	"strings"
+)
+
+// EncryptPAL encodes and encrypts the given participant DIDs as PAL header.
+// It uses the given types.KeyResolver to look up the public encryption key for each participant,
+// and then encrypts the PAL header using each.
+func EncryptPAL(keyResolver types.KeyResolver, participants []did.DID) ([][]byte, error) {
+	var encryptionKeys []*ecdsa.PublicKey
+	var recipientStrs []string
+	for _, recipient := range participants {
+		recipientStrs = append(recipientStrs, recipient.String())
+		rawKak, err := keyResolver.ResolveKeyAgreementKey(recipient)
+		if err != nil {
+			return nil, fmt.Errorf("unable to resolve keyAgreement key (recipient=%s): %w", recipient, err)
+		}
+		kak, ok := rawKak.(*ecdsa.PublicKey)
+		if !ok {
+			return nil, fmt.Errorf("resolved keyAgreement key is not an elliptic curve key (recipient=%s)", recipient)
+		}
+		encryptionKeys = append(encryptionKeys, kak)
+	}
+	plaintext := []byte(strings.Join(recipientStrs, "\n"))
```

Comment:
Nitpick: if we want a byte slice anyway we could use `recipientStrs` as a `[]byte` and use `bytes.Join` here